embassy-rp/src/gpio.rs

```diff
@@ -136,16 +136,11 @@ pub enum InterruptTrigger {
     AnyEdge,
 }
 
-impl InterruptTrigger {
-    fn from_u32(value: u32) -> Option<InterruptTrigger> {
-        match value {
-            1 => Some(InterruptTrigger::LevelLow),
-            2 => Some(InterruptTrigger::LevelHigh),
-            3 => Some(InterruptTrigger::EdgeLow),
-            4 => Some(InterruptTrigger::EdgeHigh),
-            _ => None,
-        }
-    }
+pub(crate) unsafe fn init() {
+    let irq = interrupt::IO_IRQ_BANK0::steal();
+    irq.disable();
+    irq.set_priority(interrupt::Priority::P3);
+    irq.enable();
```

Comment:
This is run on core0 only, so it enables the interrupt in core0's nvic only. When waiting on gpio from core1, the interrupts will be routed to core1 but the interrupt won't fire since it's disabled in nvic.

Reply:
yeah that should work. The other way is to route all interrupts to core0. That's what DMA is doing. I don't think it makes much of a difference.